Author: Moritz Mühlenhoff <jmm@inutil.org>
Last-Update: 2012-02-04
Forwarded: not-needed
Subject: Please enable hardened build flags
--- a/build/Makefile.linux
+++ b/build/Makefile.linux
@@ -28,7 +28,10 @@
 
 # Debug
 
-FLAGS = -O2 -Wall -DSELECT_PROBLEM
+FLAGS = `dpkg-buildflags --get CFLAGS`
+FLAGS += -Wall -DSELECT_PROBLEM
+FLAGS += `dpkg-buildflags --get CPPFLAGS`
+FLAGS += `dpkg-buildflags --get LDFLAGS`
 M="CC=$(CC) $(FLAGS)"
 D="CC=$(CC) $(FLAGS) -DDEBUG -DTHREADS"
 #FLAGS = -g
--- a/build/Makefile.freebsd
+++ b/build/Makefile.freebsd
@@ -34,7 +34,10 @@
 # Debug
 D=$(M) DEBUG=-DDEBUG
 
-FLAGS = -O2 -Wall
+FLAGS = `dpkg-buildflags --get CFLAGS`
+FLAGS += -Wall -DSELECT_PROBLEM
+FLAGS += `dpkg-buildflags --get CPPFLAGS`
+FLAGS += `dpkg-buildflags --get LDFLAGS`
 #FLAGS = -g
 
 all: 
