Author: Moritz Mühlenhoff <jmm@inutil.org>
Last-Update: 2012-02-04
Forwarded: not-needed
Subject: Please enable hardened build flags
--- a/build/Makefile.linux
+++ b/build/Makefile.linux
@@ -28,7 +28,10 @@
 
 # Debug
 
-FLAGS = -O2 -Wall -DSELECT_PROBLEM
+FLAGS = `dpkg-buildflags --get CFLAGS`
+FLAGS += -Wall -DSELECT_PROBLEM
+FLAGS += `dpkg-buildflags --get CPPFLAGS`
+FLAGS += `dpkg-buildflags --get LDFLAGS`
 M="CC=$(CC) $(FLAGS)"
 D="CC=$(CC) $(FLAGS) -DDEBUG -DTHREADS"
 #FLAGS = -g
--- a/build/Makefile.freebsd
+++ b/build/Makefile.freebsd
@@ -26,17 +26,20 @@
 M="VERSION=$(VERSION)"
 Q="DEFINES=-DSELECT_PROBLEM"
 
+FLAGS = `dpkg-buildflags --get CFLAGS`
+FLAGS += -O2 -Wall -DSELECT_PROBLEM -pipe -D_THREAD_SAFE -pthread
+FLAGS += `dpkg-buildflags --get CPPFLAGS`
+FLAGS += `dpkg-buildflags --get LDFLAGS`
+#FLAGS = -g
+
 # FreeBSD needs some special flags to compile MaraDNS
-CC="cc -O2 -Wall -pipe -D_THREAD_SAFE -pthread"
+CC="cc $(FLAGS)"
 M=CC=$(CC) 
 V="VERSION=$(VERSION)"
 
 # Debug
 D=$(M) DEBUG=-DDEBUG
 
-FLAGS = -O2 -Wall
-#FLAGS = -g
-
 all: 
 	cd libs ; make $(M) ; cd ../dns ; make $(M) ; \
 	cd ../rng ; make $(M) ; cd ../parse ; make $(M) ; \
